harpy/_validations.py

```
@@ -467,8 +487,38 @@ def check_fasta(genofile, quiet):
         print_solution(solutiontext + "\nSee the FASTA file spec and try again after making the appropriate changes: https://www.ncbi.nlm.nih.gov/genbank/fastaformat/")
         sys.exit(1)
 
+def fasta_contig_match(contigs, fasta):
+    """Checks whether a list of contigs are present in a fasta file"""
+    valid_contigs = []
+    if is_gzip(fasta):
+        gen_open = gzip.open(fasta, "rt")
+    else:
+        gen_open = open(fasta, "r")
```

Ruff

Use ternary operator `gen_open = gzip.open(fasta, 'rt') if is_gzip(fasta) else open(fasta, 'r')` instead of `if`-`else`-block

Replace `if`-`else`-block with `gen_open = gzip.open(fasta, 'rt') if is_gzip(fasta) else open(fasta, 'r')`

(SIM108)